### DISCUSSION TITLE
Warn user if function does not contain required connected signal parameters

### DISCUSSION THREAD

**Discussion by amarraff** [godotengine/godot-proposals#8651] [Engine Core]
Apologies in advance if this is already in the Godot issues, proposal issues, discussions, etc. I didn't find a related post with the wording I'm searching for. 

I've run into a semantic error a few times when I connect a function to a signal, but I forget to write the parameters of the signal in the function's parameters. This causes the function to not run when the connected signal is emitted. But, the engine doesn't warn me that I've done this, so I end up combing through my code and debug with strings until I realize that this is what's happening.

It would be great if Godot's error-detection system could recognize that you've told it to connect a function to a signal, but the function's parameters do not contain those necessary to properly receive the signal.

**Comment by AThousandShips** [comment]
Thank you for your proposal, consolidating in:
* https://github.com/godotengine/godot-proposals/issues/8982
